cli/cook/subcommands/submit.py

```diff
@@ -0,0 +1,186 @@
+    print('Enter the raw job(s) JSON (press Ctrl+D on a blank line to submit)', file=sys.stderr)
+    jobs_json = sys.stdin.read()
+    return jobs_json
+
+
+def submit(clusters, args):
+    """
+    Submits a job (or multiple jobs) to cook scheduler. Assembles a list of jobs,
+    potentially getting data from configuration, the command line, and stdin.
+    """
+    logging.debug('submit args: %s' % args)
+
+    job = args
+    raw = job.pop('raw', None)
+    command_from_command_line = job.pop('command', None)
+
+    if raw:
+        if command_from_command_line:
+            raise Exception('You cannot specify a command at the command line when using --raw/-r')
+
+        jobs_json = read_jobs_from_stdin()
+        jobs = parse_raw_job_spec(job, jobs_json)
+    else:
+        if command_from_command_line:
+            if command_from_command_line[0] == '--':
+                command_from_command_line = command_from_command_line[1:]
+            commands = [' '.join(command_from_command_line)]
+        else:
+            commands = read_commands_from_stdin()
+
+        logging.debug('subcommands: %s' % commands)
+
+        if job.get('uuid') and len(commands) > 1:
+            raise Exception('You cannot specify multiple subcommands with a single UUID')
+
+        if job.get('env'):
+            job['env'] = dict([e.split('=', maxsplit=1) for e in job['env']])
+
+        jobs = [deep_merge(job, {'command': c}) for c in commands]
+
+    for j in jobs:
+        if not j.get('uuid'):
+            j['uuid'] = str(uuid.uuid4())
+
+        if not j.get('name'):
+            j['name'] = '%s_job' % os.environ['USER']
+
+    return submit_federated(clusters, jobs)
+
+
+def valid_uuid(s):
+    """Allows argparse to flag user-provided job uuids as valid or not"""
+    try:
+        return str(uuid.UUID(s))
+    except:
+        raise argparse.ArgumentTypeError('%s is not a valid UUID' % s)
+
+
+def register(add_parser):
+    """Adds this sub-command's parser and returns the action function"""
+    submit_parser = add_parser('submit', help='create job for command')
+    submit_parser.add_argument('--uuid', '-u', help='uuid of job', type=valid_uuid)
+    submit_parser.add_argument('--name', '-n', help='name of job')
+    submit_parser.add_argument('--priority', '-p', help='priority of job, between 0 and 100 (inclusive), with 100 '
+                                                        'being highest priority (default = 50)',
+                               type=int, choices=range(0, 101), metavar='')
+    submit_parser.add_argument('--max-retries', help='maximum retries for job',
+                               dest='max-retries', type=int, metavar='COUNT')
+    submit_parser.add_argument('--max-runtime', help='maximum runtime for job',
+                               dest='max-runtime', type=int, metavar='MILLIS')
+    submit_parser.add_argument('--cpus', help='cpus to reserve for job', type=float)
```

Comment:
Should we have more short options?  `-c` for cpus and `-m` for mem?  Maybe also `-e` for env?

Reply:
Done